### PR TITLE
SO-2875 Implement IconIdProvider interface

### PR DESCRIPTION
**Pull request by mucst**

### CODE REVIEW

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/api/component/IconIdProvider.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2011-2018 B2i Healthcare Pte Ltd, http://b2i.sg
```
Comment (cmark) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/api/component/IconIdProvider.java]: New files should indicate only the current year in their license header. Update it, please!

**Review by cmark — CHANGES_REQUESTED** [review]

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/api/component/IconIdProvider.java (hunk 1 above)
Comment (mucst) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/api/component/IconIdProvider.java]: updated.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedComponent.java [hunk 2]
```diff
@@ -65,6 +66,8 @@ public String getModuleId() {
 	 * @beta - this method is subject to changes or even removal in future releases.  
 	 * @return - the icon ID associated with this component
 	 */
+	@Override
+	@JsonIgnore
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/domain/SnomedComponent.java:70]: This annotation was removed in a recent commit. I'll remove it after merging this PR to 6.x.

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**